features/FEATURE_BLE/source/generic/GenericSecurityManager.cpp

```diff
@@ -519,17 +556,11 @@ ble_error_t GenericSecurityManager::setOOBDataUsage(
     cb->attempt_oob = useOOB;
     cb->oob_mitm_protection = OOBProvidesMITM;
 
-    _oob_temporary_key_creator_address = cb->local_address;
-    get_random_data(_oob_temporary_key.buffer(), 16);
-
-    eventHandler->legacyPairingOobGenerated(
-        &_oob_temporary_key_creator_address,
-        &_oob_temporary_key
-    );
-
-    _pal.generate_secure_connections_oob(connection);
-
-    return BLE_ERROR_NONE;
+    if (useOOB) {
```

Comment:
What if OOB has already been generated and exchanged ?

Reply:
Then you don't have to call this function.